rust/sealevel/programs/interchain-accounts/src/processor.rs

```diff
@@ -0,0 +1,507 @@
+            program_id,
+            dispatch_authority_seeds,
+            remote_call.destination,
+            encoded_message.into(),
+            remote_call.gas_limit.unwrap_or(DEFAULT_HANDLE_GAS_AMOUNT),
+            dispatch_account_metas,
+            dispatch_account_infos,
+            igp_payment_account_metas,
+            &igp_payment_account_infos,
+        )?;
+    } else {
+        // Dispatch the message.
+        storage.dispatch(
+            program_id,
+            dispatch_authority_seeds,
+            remote_call.destination,
+            encoded_message.into(),
+            dispatch_account_metas,
+            dispatch_account_infos,
+        )?;
+    }
+
+    msg!(
+        "Remote call requested, owner: {}, destination: {}, ism: {:?}, salt: {:?}",
+        owner,
+        remote_call.destination,
+        remote_call.ism,
+        remote_call.salt
+    );
+
+    Ok(())
+}
+
+/// Accounts:
+/// 0. `[writeable]` Storage PDA account.
+/// 1. `[signer]` Owner.
+fn set_interchain_security_module(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    ism: Option<Pubkey>,
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+
+    // Account 0: Storage PDA account.
+    // Not bothering with validity checks because this is a test program
+    let storage_info = next_account_info(accounts_iter)?;
+    let (expected_storage_pda_key, _expected_storage_pda_bump) =
+        Pubkey::find_program_address(program_storage_pda_seeds!(), program_id);
+    if storage_info.key != &expected_storage_pda_key {
+        return Err(ProgramError::InvalidArgument);
+    }
+    let mut storage =
+        InterchainAccountStorageAccount::fetch(&mut &storage_info.data.borrow()[..])?.into_inner();
+
+    // Account 1: Owner.
+    let owner_info = next_account_info(accounts_iter)?;
+    storage.ensure_owner_signer(owner_info)?;
+
+    storage.ism = ism;
+
+    // Store it
+    InterchainAccountStorageAccount::from(storage).store(storage_info, false)?;
+
+    Ok(())
+}
+
+/// Accounts:
+/// 0. `[]` Storage PDA account.
+fn get_interchain_security_module(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+
+    // Account 0: Storage PDA account.
+    let storage_info = next_account_info(accounts_iter)?;
+    let (expected_storage_pda_key, _expected_storage_pda_bump) =
+        Pubkey::find_program_address(program_storage_pda_seeds!(), program_id);
+    if storage_info.key != &expected_storage_pda_key {
+        return Err(ProgramError::InvalidArgument);
+    }
+    let storage =
+        InterchainAccountStorageAccount::fetch(&mut &storage_info.data.borrow()[..])?.into_inner();
+
+    set_return_data(
+        &storage
+            .ism
+            .try_to_vec()
+            .map_err(|err| ProgramError::BorshIoError(err.to_string()))?[..],
+    );
+
+    Ok(())
+}
+
+fn set_account_meta_return_data(program_id: &Pubkey) -> ProgramResult {
```

Comment:
nit: maybe better to name this something specific to the ISM account metas, as I imagine if/when this has a handle fn we'll have the account metas for that be very different